lib/facility.rb

```diff
@@ -1,14 +1,49 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
   def initialize(facility_information)
     @name = facility_information[:name]
     @address = facility_information[:address]
     @phone = facility_information[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    return if !self.services.include?("Vehicle Registration")
+    vehicle.registration_date = Date.today
+    @registered_vehicles << vehicle
+    set_plate_type(vehicle)
+    collect_fees(vehicle)
+  end
+
+  def set_plate_type(vehicle)
+    if vehicle.electric_vehicle?
+      vehicle.plate_type = :ev
+    elsif vehicle.antique?
+      vehicle.plate_type = :antique
+    else 
+      vehicle.plate_type = :regular
+    end
```

Comment:
For funsies if you have time after you finish your project:
This method works because you are using an `attr_accessor` in your vehicle class. If you removed the `attr_accessor`, how would you have to rewrite your method(s) to do the same thing? 

Whichever way you do it will end up your own preference (attr_accessor or none) but learning how to do it both ways could be a good exercise.

Reply:
Noticed your own comment on the pull request regarding using the assignment operator. That actually is another way of doing the attr_accessor which has the same problems that some people don't like in attr_accessors.

Writing a method to change an attribute without an attr_accessor will actually be pretty similar to other methods you've written to change an attribute, like the `earn_permit` method. 
``` ruby
def earn_permit
    @permit = true 
  end
```
Similarly, something like
``` ruby
def add_registration_date
   @registration_date = Date.today
  end
```
In your vehicle class will do what you are looking for! Then you'd call `vehicle.add_registration_date` from your `facility` class when you register it.